pkg/domain/proxy.go

```diff
@@ -0,0 +1,25 @@
+package domain
+
+import "strings"
+
+type ProxyConfiguration struct {
+	HTTPProxy  string
+	HTTPSProxy string
+
+	/* no_proxy is a comma- or space-separated list of machine
+	   or domain names, with optional :port part.  If no :port
+	   part is present, it applies to all ports on that domain.
+	*/
+	NoProxy string
+}
+
+func (p ProxyConfiguration) GetURLForEndpoint(endpoint string) string {
+	if !strings.Contains(p.NoProxy, endpoint) {
```

Comment:
That's a weird condition, I don't think we are supposed to do any logic in here. Those values are mean to be passed down the stack, its all part of `golang.org/x/net/http/httpproxy` [spec](https://pkg.go.dev/golang.org/x/net/http/httpproxy#Config)